packages/block-library/src/quote/transforms.js

```diff
@@ -1,71 +1,150 @@
 /**
  * WordPress dependencies
  */
-import { createBlock } from '@wordpress/blocks';
-import { create, join, split, toHTMLString } from '@wordpress/rich-text';
+import {
+	createBlock,
+	parseWithAttributeSchema,
+	rawHandler,
+	serialize,
+	switchToBlockType,
+} from '@wordpress/blocks';
+
+const toBlocksOfType = ( blocks, type ) => {
+	const result = [];
+	blocks.forEach( ( block ) => {
+		if ( type === block.name ) {
+			return result.push( block );
+		}
+		const newBlocks = switchToBlockType( block, type );
+		if ( newBlocks ) {
+			result.push( ...newBlocks.filter( Boolean ) );
+		}
+	} );
+	return result;
+};
+
+const unwrapContainers = ( blocks ) =>
+	blocks.length === 1 && blocks[ 0 ].name === 'core/group'
+		? blocks[ 0 ].innerBlocks
+		: blocks;
 
 const transforms = {
 	from: [
+		{
+			type: 'block',
+			isMultiBlock: true,
+			blocks: [ '*' ],
+			__experimentalConvert: ( blocks ) =>
+				createBlock(
+					'core/quote',
+					{},
+					unwrapContainers( blocks ).map( ( block ) =>
+						createBlock(
+							block.name,
+							block.attributes,
+							block.innerBlocks
+						)
+					)
+				),
+		},
 		{
 			type: 'block',
 			isMultiBlock: true,
 			blocks: [ 'core/paragraph' ],
 			transform: ( attributes ) => {
-				return createBlock( 'core/quote', {
-					value: toHTMLString( {
-						value: join(
-							attributes.map( ( { content } ) =>
-								create( { html: content } )
-							),
-							'\u2028'
-						),
-						multilineTag: 'p',
-					} ),
-					anchor: attributes.anchor,
-				} );
+				return createBlock(
+					'core/quote',
+					{},
+					attributes.map( ( props ) =>
+						createBlock( 'core/paragraph', props )
+					)
+				);
 			},
 		},
 		{
 			type: 'block',
+			isMultiBlock: true,
 			blocks: [ 'core/heading' ],
-			transform: ( { content, anchor } ) => {
-				return createBlock( 'core/quote', {
-					value: `<p>${ content }</p>`,
-					anchor,
-				} );
+			transform: ( attributes ) => {
+				return createBlock(
+					'core/quote',
+					{},
+					attributes.map( ( props ) =>
+						createBlock( 'core/heading', props )
+					)
+				);
 			},
 		},
 		{
 			type: 'block',
 			blocks: [ 'core/pullquote' ],
```

Comment:
Shouldn't we just update this very similar block as part of the same PR?

Reply:
If that is true, perhaps we should simplify pull quote to not allow multiple paragraphs?